### ISSUE TITLE
Pure Market Making: orders not cancelled due to network errors creates unwanted hanging orders behaviour

### ISSUE THREAD

**Issue by zappra** [CoinAlpha/hummingbot#3374]
**Describe the bug**
It's possible for Hummingbot to lose track of orders created, due to network errors. These orders however persist on the exchange creating an unwanted hanging orders behaviour

Ordinarily max_order_age would cancel any orders over the default of 30 mins but since these orders are "missing" they haven't been picked up.

**Steps To Reproduce**
This bug appears to be caused by failure to handle or recover from network errors although it might be possible to repro in test

There are 6 or so instances in the log, but for example:

00:57:56 - limit buy order created, 22.370 AVAX @ 34.550

this is immediately followed by some network status change/error messages

The next order cycle occurs at 01:22:57 where another limit buy order is created without cancelling the first one

At this point the first order is apparently not being tracked internally but is filled on the exchange some time later, and picked up at 8:56:01 with INFO - Recreating missing trade in TradeFill

The net result was the quick execution of 6 "dead" buy orders resulting in the accumulation of way more base asset than intended

**Release version**
dev 0.39.0

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 

[logs_conf_pure_mm_avax-usdt.log](https://github.com/CoinAlpha/hummingbot/files/6478487/logs_conf_pure_mm_avax-usdt.log)

┆Issue is synchronized with this [Clickup task](https://app.clickup.com/t/1ramjxv) by [Unito](https://www.unito.io)

**Comment by phbrgnomo** [comment]
Were you using `hanging_orders`?

**Comment by zappra** [comment]
No single order mode, hanging orders disabled

**Comment by krisj** [comment]
this sounds nearly exactly like #3341

**Comment by zappra** [comment]
@phbrgnomo is there any update on this issue, eg. is there some WIP in a branch somewhere?

**Comment by nikspz** [comment]
@zappra  we don't have an estimated time yet when it will be fixed but rest assured that the team is working to fix this as soon as possible. We apologize for any inconvenience this may have caused.
You could help us improve Hummingbot Development by Upvote the standing issues on the Github to help our team decide what fixes we should prioritize!
https://hummingbot.io/blog/2021-06-help-us-improve-hummingbot-vote-for-bugs

**Comment by petioptrv** [comment]
@zappra, I'm trying to reproduce the issue you're experiencing by cutting my connection to simulate a network disconnect, but I can't seem to get it to drop the order. Can you share an example of a config that has generated this error for you in the past?

**Comment by zappra** [comment]
@petioptrv very simple config, single order mode, 2% spread, 25 minute order refresh

I've not been able to reproduce the error locally either, and haven't seen it now in live trading (as of Aug 24) for some time now, since end of May, when markets were particularly volatile.

**Comment by dennisocana** [comment]
@CoinAlpha/qa-team @RC-13 please re-attempt to reproduce this bug and close if needed.

**Comment by rapcmia** [comment]
Hi @zappra, I tried to reproduce this by using the same 2% spread and refresh time of 25mins on doge-usdt and eth-usdt pairs on Binance. Tested it locally by being disconnected for 2-3mins and also while near the refresh time. The bot seems to be able to retrieve and recreate orders ok. Just checking with you again if this is still an ongoing issue? Else we can have this ticket closed for now. Thank you
